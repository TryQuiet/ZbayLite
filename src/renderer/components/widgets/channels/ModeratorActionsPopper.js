import React from 'react'
import PropTypes from 'prop-types'
import * as R from 'ramda'

import {
  Popper,
  Fade,
  Paper,
  withStyles,
  Grid,
  Typography
} from '@material-ui/core'

import Icon from '../../ui/Icon'
import userIcon from '../../../static/images/userIcon.svg'
import banIcon from '../../../static/images/banIcon.svg'
import removeMessageIcon from '../../../static/images/removeMessageIcon.svg'

const styles = theme => ({
  root: {
    width: 320,
    height: 150,
    borderRadius: 8,
    backgroundColor: theme.palette.colors.white,
    boxShadow: '0px 2px 25px rgba(0, 0, 0, 0.2)'
  },
  username: {
    lineHeight: '26px',
    fontWeight: 500,
    marginBottom: -3
  },
  user: {
    marginTop: 10,
    marginRight: 12
  },
  address: {
    color: theme.palette.colors.captionPurple,
    lineHeight: '18px',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    maxWidth: 230,
    whiteSpace: 'nowrap',
    display: 'inline-block'
  },
  info: {
    padding: 10,
    height: 60,
    borderBottom: '1px solid #DBDBDB'
  },
  action: {},
  actions: {
    paddingLeft: 15,
    paddingRight: 15,
    paddingTop: 14
  },
  removeIcon: {
    marginRight: 10,
    marginTop: 5
  },
  banIcon: {
    marginRight: 6,
    marginTop: 4
  },
  banDiv: { marginTop: 9, cursor: 'pointer' },
  pointer: {
    cursor: 'pointer'
  }
})
export const ModeratorActionsPopper = ({
  classes,
  name,
  address,
  open,
  anchorEl,
  banUser,
  removeMessage
}) => (
  <Popper
    open={open}
    anchorEl={anchorEl}
    anchorOrigin={{
      vertical: 'bottom',
      horizontal: 'center'
    }}
    transformOrigin={{
      vertical: 'top',
      horizontal: 'center'
    }}
    transition
  >
    {({ TransitionProps }) => (
      <Fade {...TransitionProps} timeout={350}>
        <Paper>
          <Typography className={classes.typography}>
            The content of the Popper.
          </Typography>
        </Paper>
      </Fade>
    )}
    <Grid container direction='column' className={classes.root}>
      <Grid item container direction='row' className={classes.info} spacing={0}>
        <Grid item>
          <Icon className={classes.user} src={userIcon} />
        </Grid>
        <Grid item>
          <Grid item xs={12}>
            <Typography variant='h5' className={classes.username}>
              {name}
            </Typography>
          </Grid>
          <Grid item xs={12}>
            <Typography variant='caption' className={classes.address}>
              {address}
            </Typography>
          </Grid>
        </Grid>
      </Grid>
      <Grid
        item
        container
        direction='column'
        justify='space-between'
        alignItems='center'
        className={classes.actions}
      >
        {/* <Grid // Temporarily hidden
          item
          container
          onClick={() => {
            removeMessage()
          }}
          className={classes.pointer}
        >
          <Grid item>
            <Icon className={classes.removeIcon} src={removeMessageIcon} />
          </Grid>
          <Grid item xs>
            <Typography variant='body1' className={classes.action}>
              Hide message
            </Typography>
          </Grid>
        </Grid> */}
        <Grid
          item
          container
          onClick={() => {
            banUser()
          }}
          className={classes.banDiv}
        >
          <Grid item>
            <Icon className={classes.banIcon} src={banIcon} />
          </Grid>
          <Grid item xs>
            <Typography variant='body1' className={classes.action}>
              Silence user
            </Typography>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  </Popper>
)

ModeratorActionsPopper.propTypes = {
  classes: PropTypes.object.isRequired,
  anchorEl: PropTypes.object.isRequired,
  name: PropTypes.string.isRequired,
  address: PropTypes.string.isRequired,
  open: PropTypes.bool.isRequired,
  removeMessage: PropTypes.func.isRequired,
  banUser: PropTypes.func.isRequired
}

export default R.compose(React.memo, withStyles(styles))(ModeratorActionsPopper)
