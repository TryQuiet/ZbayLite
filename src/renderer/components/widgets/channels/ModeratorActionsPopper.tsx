import React from 'react'

import {
  Popper,
  Fade,
  Paper,
  makeStyles,
  Grid,
  Typography
} from '@material-ui/core'
import { PopperProps } from '@material-ui/core/Popper'

import Icon from '../../ui/Icon/Icon'
import userIcon from '../../../static/images/userIcon.svg'
import banIcon from '../../../static/images/banIcon.svg'

const useStyles = makeStyles((theme) => ({
  root: {
    width: 320,
    height: 150,
    borderRadius: 8,
    backgroundColor: theme.palette.colors.white,
    boxShadow: '0px 2px 25px rgba(0, 0, 0, 0.2)'
  },
  username: {
    lineHeight: '26px',
    fontWeight: 500,
    marginBottom: -3
  },
  user: {
    marginTop: 10,
    marginRight: 12
  },
  address: {
    color: theme.palette.colors.captionPurple,
    lineHeight: '18px',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    maxWidth: 230,
    whiteSpace: 'nowrap',
    display: 'inline-block'
  },
  info: {
    padding: 10,
    height: 60,
    borderBottom: '1px solid #DBDBDB'
  },
  action: {},
  actions: {
    paddingLeft: 15,
    paddingRight: 15,
    paddingTop: 14
  },
  removeIcon: {
    marginRight: 10,
    marginTop: 5
  },
  banIcon: {
    marginRight: 6,
    marginTop: 4
  },
  banDiv: { marginTop: 9, cursor: 'pointer' },
  pointer: {
    cursor: 'pointer'
  }
}))

interface ModeratorActionsPopperProps {
  name: string
  address: string
  open: boolean
  anchorEl: PopperProps['anchorEl']
  banUser: () => void
}

export const ModeratorActionsPopper: React.FC<ModeratorActionsPopperProps> = ({
  name,
  address,
  open,
  anchorEl,
  banUser
}) => {
  const classes = useStyles({})
  return (
    <Popper
      open={open}
      anchorEl={anchorEl}
      transition
    >
      {({ TransitionProps }) => (
        <Fade {...TransitionProps} timeout={350}>
          <Paper>
            <Typography >
              The content of the Popper.
            </Typography>
          </Paper>
        </Fade>
      )}
      <Grid container direction='column' className={classes.root}>
        <Grid item container direction='row' className={classes.info} spacing={0}>
          <Grid item>
            <Icon className={classes.user} src={userIcon} />
          </Grid>
          <Grid item>
            <Grid item xs={12}>
              <Typography variant='h5' className={classes.username}>
                {name}
              </Typography>
            </Grid>
            <Grid item xs={12}>
              <Typography variant='caption' className={classes.address}>
                {address}
              </Typography>
            </Grid>
          </Grid>
        </Grid>
        <Grid
          item
          container
          direction='column'
          justify='space-between'
          alignItems='center'
          className={classes.actions}
        >
          <Grid
            item
            container
            onClick={() => {
              banUser()
            }}
            className={classes.banDiv}
          >
            <Grid item>
              <Icon className={classes.banIcon} src={banIcon} />
            </Grid>
            <Grid item xs>
              <Typography variant='body1' className={classes.action}>
                Silence user
              </Typography>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Popper>
  )
}

export default ModeratorActionsPopper
