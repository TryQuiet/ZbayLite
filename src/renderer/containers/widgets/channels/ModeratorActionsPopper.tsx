import React from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import ModeratorActionsPopperComponent from '../../../components/widgets/channels/ModeratorActionsPopper'

export const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      banUser: () => { },
      // moderationHandlers.epics.handleModerationAction({
      //   moderationType: moderationActionsType.BLOCK_USER,
      //   moderationTarget: publicKey
      // }
      removeMessage: () => { }
      // moderationHandlers.epics.handleModerationAction({
      //   moderationType: moderationActionsType.REMOVE_MESSAGE,
      //   moderationTarget: txid
      // })
    },
    dispatch
  )
}
export const ModeratorActionsPopper = ({ name, address, open, anchorEl, banUser }) => {
  return (
    <ModeratorActionsPopperComponent
      name={name}
      address={address}
      open={open}
      anchorEl={anchorEl}
      banUser={banUser}
    />
  )
}

export default connect(null, mapDispatchToProps)(ModeratorActionsPopper)
