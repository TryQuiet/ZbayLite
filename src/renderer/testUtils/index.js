import { DateTime, Settings } from 'luxon'
import secp256k1 from 'secp256k1'
import BigNumber from 'bignumber.js'
import { hash } from '../zbay/messages'
import defaultChannels from '../zcash/channels'
import { messageType } from '../../shared/static'
Settings.defaultZoneName = 'utc'

export const vaultTestMessages = [
  {
    id: '123',
    properties: {
      title: 'test',
      createdAt: '123456',
      message: '{"itemId":"6b31f1a5c68902a767eb542fa17daeb338e32d12704ac124ce55994754a5001e","text":"hello"}',
      sender: 'testsender',
      senderUsername: 'testusername',
      type: '11',
      id: '049a7d2ab30817765ea9875bfca40299669e1631329fc58b2a053ff58f78b007'
    }
  },
  {
    id: '1234',
    properties: {
      title: 'test2',
      createdAt: '123456',
      message: '{"itemId":"6b31f1a5c68902a767eb542fa17daeb338e32d12704ac124ce55994754a5001e","text":"hello"}',
      sender: 'testsender',
      senderUsername: 'testusername',
      type: '11',
      id: '149a7d2ab30817765ea9875bfca40299669e1631329fc58b2a053ff58f78b007'
    }
  }
]
export const identities = [
  {
    address: 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya',
    username: 'Wenus'
  },
  {
    address: 'zs1dhqp9dtr4pksnmaynp2k22qduvywejg3neqq4swd4a6gnz6w0m208kefcdm9n2067yn5clcvgsq',
    username: 'Mars'
  }
]

const privateKey = 'ceea41a1c9e91f839c96fba253b620da70992954b2a28b19322b191d8f5e56db'
const pKey = Buffer.alloc(32)
pKey.write(privateKey, 0, 'hex')

export const now = DateTime.utc(2019, 3, 7, 13, 3, 48)

export const createChannel = id => ({
  id: id,
  name: `Channel ${id}`,
  private: Boolean(id % 2),
  address: `zs1testaddress${id}`,
  unread: 12,
  description: id % 2 === 0 ? '' : `Channel about ${id}`,
  advertFee: 0.15,
  keys: {
    ivk: `incoming-viewing-key-${id}`,
    sk: 'test-secret-key'
  }
})

export const createUsersChannel = (id, network = 'testnet') => {
  return {
    id: id,
    name: `Channel ${id}`,
    private: Boolean(id % 2),
    address: defaultChannels.registeredUsers[network].address,
    unread: id,
    description: id % 2 === 0 ? '' : `Channel about ${id}`,
    keys: {
      ivk: `incoming-viewing-key-${id}`
    }
  }
}

export const channels = {
  createChannel,
  createUsersChannel
}

export const createVaultMessage = (
  id,
  createdAt = now.minus({ hours: id }).toSeconds(),
  replyTo = 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya'
) => ({
  id,
  status: 'broadcasted',
  spent: new BigNumber('123'),
  type: messageType.BASIC,
  createdAt,
  message: `This is some message ${id}`,
  sender: {
    replyTo,
    username: 'test'
  }
})

export const createMessage = (id, createdAt = now.minus({ hours: id }).toSeconds()) => ({
  id,
  type: messageType.BASIC,
  createdAt,
  message: `This is some message ${id}`,
  signature: secp256k1.sign(hash(`This is a message with id ${id}`), pKey).signature
})
export const createItemMessage = (id, createdAt = now.minus({ hours: id }).toSeconds()) => ({
  id,
  type: messageType.BASIC,
  createdAt,
  message: { itemId: 'test1', test: `This is some message ${id}` },
  signature: secp256k1.sign(hash(`This is a message with id ${id}`), pKey).signature
})

export const createReceivedMessage = ({
  id,
  createdAt = now.toSeconds(),
  sender = identities[0]
}) => ({
  id,
  spent: new BigNumber('0.32'),
  type: messageType.BASIC,
  createdAt,
  message: `This is a message with id ${id}`,
  signature: secp256k1.sign(hash(`This is a message with id ${id}`), pKey).signature
})

export const createSendableMessage = ({
  message,
  createdAt = now.toSeconds(),
  type = messageType.BASIC
}) => ({
  type,
  signature: secp256k1.sign(hash(message), pKey).signature,
  createdAt,
  message
})

export const createSendableUserMessage = ({
  message,
  createdAt = now.toSeconds(),
  type = messageType.USER
}) => ({
  type,
  signature: secp256k1.sign(hash(JSON.stringify(message) + createdAt), pKey).signature,
  createdAt,
  message
})

export const createSendableTransferMessage = ({
  message = 'hello',
  createdAt = now.toSeconds()
}) => ({
  type: messageType.TRANSFER,
  spent: new BigNumber('0.32'),
  sender: {
    replyTo: 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya',
    username: 'Wenus'
  },
  receiver: {
    replyTo: 'zs1z9rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya',
    username: 'Sun'
  },
  createdAt,
  signature: secp256k1.sign(hash(message), pKey).signature,
  message
})

export const createReceivedTransferMessage = ({ id, createdAt = now.toSeconds() }) => ({
  id,
  spent: new BigNumber('0.32'),
  type: messageType.TRANSFER,
  createdAt,
  message: `This is a message with id ${id}`,
  signature: secp256k1.sign(hash(`This is a message with id ${id}`), pKey).signature
})

export const createReceivedUserMessage = ({ id, createdAt = now.toSeconds() }) => ({
  id,
  spent: new BigNumber('0.32'),
  type: messageType.USER,
  createdAt,
  message: {
    firstName: 'testname',
    lastName: 'testlastname',
    nickname: 'nickname',
    address:
      'ztestsapling14dxhlp8ps4qmrslt7pcayv8yuyx78xpkrtfhdhae52rmucgqws2zp0zwf2zu6qxjp96lzapsn4r'
  },
  signature: secp256k1.sign(
    hash(
      JSON.stringify({
        firstName: 'testname',
        lastName: 'testlastname',
        nickname: 'nickname',
        address:
          'ztestsapling14dxhlp8ps4qmrslt7pcayv8yuyx78xpkrtfhdhae52rmucgqws2zp0zwf2zu6qxjp96lzapsn4r'
      }) + id
    ),
    pKey
  ).signature,
  r: secp256k1.sign(
    hash(
      JSON.stringify({
        firstName: 'testname',
        lastName: 'testlastname',
        nickname: 'nickname',
        address:
          'ztestsapling14dxhlp8ps4qmrslt7pcayv8yuyx78xpkrtfhdhae52rmucgqws2zp0zwf2zu6qxjp96lzapsn4r'
      }) + id
    ),
    pKey
  ).recovery
})

export const messages = {
  createReceivedMessage,
  createSendableMessage,
  createReceivedTransferMessage,
  createSendableTransferMessage,
  createMessage,
  createVaultMessage,
  createReceivedUserMessage,
  createSendableUserMessage
}

export const createTransfer = ({ txid, amount = '0', memo = '', change = false, blockHeight = 100 }) => ({
  txid,
  amount,
  memo,
  memohex: memo,
  change,
  blockHeight: 100
})

export const transfers = {
  createTransfer
}

export const createIdentity = ({
  name = 'Saturn',
  address = 'saturn-private-address',
  transparentAddress = 'saturn-transparent-address',
  signerPrivKey = 'AyrXTov+4FEyle1BrxndDhePlUOB1nPV5YvyexC8m6E=',
  signerPubKey = 'Alw9G29ahXm5et9T7FJF3lbXVXNkapR826yPtfMVJYnK',
  keys = {
    tpk: 'saturn-tpk',
    sk: 'saturn-sk'
  },
  shippingData = {
    firstName: 'Saturn',
    lastName: 'the Planet',
    street: 'Coders Dv',
    country: 'Poland',
    region: 'Malopolska',
    city: 'Krakow',
    postalCode: '1337-455'
  }
} = {}) => ({
  name,
  address,
  transparentAddress,
  signerPrivKey,
  signerPubKey,
  shippingData,
  keys
})

export default {
  identities,
  now,
  transfers,
  channels,
  messages,
  createIdentity,
  createMessage,
  createSendableTransferMessage,
  createItemMessage,
  createReceivedMessage,
  vaultTestMessages
}
