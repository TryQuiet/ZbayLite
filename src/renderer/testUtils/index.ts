import { DateTime, Settings } from 'luxon'
import BigNumber from 'bignumber.js'
import { messageType } from '../../shared/static'

Settings.defaultZoneName = 'utc'

export const now = DateTime.utc(2019, 3, 7, 13, 3, 48)

export const identities = [
  {
    address: 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya',
    username: 'Wenus'
  },
  {
    address: 'zs1dhqp9dtr4pksnmaynp2k22qduvywejg3neqq4swd4a6gnz6w0m208kefcdm9n2067yn5clcvgsq',
    username: 'Mars'
  }
]

export const createChannel = (id: number) => ({
  id: id,
  name: `Channel ${id}`,
  private: Boolean(id % 2),
  address: `zs1testaddress${id}`,
  unread: 12,
  description: id % 2 === 0 ? '' : `Channel about ${id}`,
  advertFee: 0.15,
  keys: {
    ivk: `incoming-viewing-key-${id}`,
    sk: 'test-secret-key'
  }
})

export const channels = {
  createChannel
}

export const createVaultMessage = (
  id,
  createdAt = now.minus({ hours: id }).toSeconds(),
  replyTo = 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya'
) => ({
  id,
  status: 'broadcasted',
  spent: new BigNumber('123'),
  type: messageType.BASIC,
  createdAt,
  message: `This is some message ${id}`,
  sender: {
    replyTo,
    username: 'test'
  }
})

export const createMessage = async () => {
  return {
    id: '1',
    type: messageType.BASIC,
    message: 'example message',
    pubKey: 'ownPubKey',
    channelId: 'address',
    createdAt: 123445789,
    signature: 'secretsignature'
  }
}

export const messages = {
  createMessage,
  createVaultMessage
}

export const createIdentity = ({
  name = 'Saturn',
  address = 'saturn-private-address',
  transparentAddress = 'saturn-transparent-address',
  signerPrivKey = 'AyrXTov+4FEyle1BrxndDhePlUOB1nPV5YvyexC8m6E=',
  signerPubKey = 'Alw9G29ahXm5et9T7FJF3lbXVXNkapR826yPtfMVJYnK',
  keys = {
    tpk: 'saturn-tpk',
    sk: 'saturn-sk'
  },
  shippingData = {
    firstName: 'Saturn',
    lastName: 'the Planet',
    street: 'Coders Dv',
    country: 'Poland',
    region: 'Malopolska',
    city: 'Krakow',
    postalCode: '1337-455'
  }
} = {}) => ({
  name,
  address,
  transparentAddress,
  signerPrivKey,
  signerPubKey,
  shippingData,
  keys
})

export default {
  identities,
  now,
  channels,
  messages,
  createIdentity,
  createMessage
}
