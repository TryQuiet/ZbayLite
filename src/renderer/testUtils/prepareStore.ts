import { communities, identity, users, errors, messages, publicChannels } from '@zbayapp/nectar'
import { StoreKeys as NectarStoreKeys } from '@zbayapp/nectar/lib/sagas/store.keys'
import { StoreKeys } from '../store/store.keys'
import { combineReducers, createStore, applyMiddleware } from 'redux'
import createSagaMiddleware from 'redux-saga'
import thunk from 'redux-thunk'
import rootSaga from '../sagas/index.saga'
import { socketActions, socketReducer } from '../sagas/socket/socket.slice'
import { modalsReducer } from '../sagas/modals/modals.slice'
import MockedSocket from 'socket.io-mock'
import { fork, delay, call, put } from 'typed-redux-saga'

export const reducers = {
  [NectarStoreKeys.Communities]: communities.reducer,
  [NectarStoreKeys.Identity]: identity.reducer,
  [NectarStoreKeys.Users]: users.reducer,
  [NectarStoreKeys.Errors]: errors.reducer,
  [NectarStoreKeys.Messages]: messages.reducer,
  [NectarStoreKeys.PublicChannels]: publicChannels.reducer,
  [StoreKeys.Socket]: socketReducer,
  [StoreKeys.Modals]: modalsReducer
}

export const prepareStore = async (
  mockedState?: { [key in StoreKeys | NectarStoreKeys]?: any },
  mockedSocket?: MockedSocket
) => {
  const combinedReducers = combineReducers(reducers)
  const sagaMiddleware = createSagaMiddleware()
  const store = createStore(
    combinedReducers,
    mockedState,
    applyMiddleware(...[sagaMiddleware, thunk])
  )
  // Fork Nectar's sagas (require mocked socket.io-client)
  if (mockedSocket) {
    sagaMiddleware.run(rootSaga)
    // Mock socket connected event
    await sagaMiddleware.run(mockSocketConnectionSaga, mockedSocket).toPromise()
  }
  return {
    store,
    runSaga: sagaMiddleware.run
  }
}

function* mockSocketConnectionSaga(socket: MockedSocket): Generator {
  yield* fork(function* (): Generator {
    yield* delay(1000)
    yield* call(() => {
      socket.socketClient.emit('connect')
    })
  })
  yield* put(socketActions.startConnection({ dataPort: 4677 }))
}
