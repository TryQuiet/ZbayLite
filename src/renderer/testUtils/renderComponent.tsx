import React, { FC, ReactElement } from 'react'
import { MuiThemeProvider } from '@material-ui/core'
import { Provider } from 'react-redux'

import { render } from '@testing-library/react'

import theme from '../theme'
import store from '../store'
import { Store } from 'redux'

export const renderComponent = (ui: ReactElement, storeState: Store = store): ReturnType<typeof render> => {
  const Wrapper: FC = ({ children }) => (
    <MuiThemeProvider theme={theme}>
      <Provider store={storeState}>
        {children}
      </Provider>
    </MuiThemeProvider>
  )

  return render(ui, { wrapper: Wrapper })
}
