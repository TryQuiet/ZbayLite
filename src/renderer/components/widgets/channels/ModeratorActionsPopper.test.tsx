import React from 'react'

import { ModeratorActionsPopper } from './ModeratorActionsPopper'
import { renderComponent } from '../../../testUtils/renderComponent'

describe('ModeratorActionsPopper', () => {
  it('renders', () => {
    const ref = React.createRef<HTMLAnchorElement>()
    const result = renderComponent(
      <ModeratorActionsPopper
        name='123'
        address='123'
        open
        anchorEl={ref.current}
        banUser={jest.fn()}
      />
    )
    expect(result.baseElement).toMatchInlineSnapshot(`
      <body>
        <div />
        <div
          role="tooltip"
          style="position: fixed; top: 0px; left: 0px;"
        >
          <div
            class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-direction-xs-column"
          >
            <div
              class="MuiGrid-root makeStyles-info-5 MuiGrid-container MuiGrid-item"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <img
                  class="makeStyles-user-3"
                  src="test-file-stub"
                />
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <h5
                    class="MuiTypography-root makeStyles-username-2 MuiTypography-h5"
                  >
                    123
                  </h5>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <span
                    class="MuiTypography-root makeStyles-address-4 MuiTypography-caption"
                  >
                    123
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root makeStyles-actions-7 MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
            >
              <div
                class="MuiGrid-root makeStyles-banDiv-10 MuiGrid-container MuiGrid-item"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <img
                    class="makeStyles-banIcon-9"
                    src="test-file-stub"
                  />
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
                >
                  <p
                    class="MuiTypography-root makeStyles-action-6 MuiTypography-body1"
                  >
                    Silence user
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </body>
    `)
  })
})
