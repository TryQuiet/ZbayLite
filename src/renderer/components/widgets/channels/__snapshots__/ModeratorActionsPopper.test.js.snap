// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModeratorActionsPopper renders 1`] = `
<ForwardRef(Popper)
  anchorEl={
    Object {
      "current": null,
    }
  }
  anchorOrigin={
    Object {
      "horizontal": "center",
      "vertical": "bottom",
    }
  }
  open={true}
  transformOrigin={
    Object {
      "horizontal": "center",
      "vertical": "top",
    }
  }
  transition={true}
>
  <Component />
  <WithStyles(ForwardRef(Grid))
    className="root-mock"
    container={true}
    direction="column"
  >
    <WithStyles(ForwardRef(Grid))
      className="info-mock"
      container={true}
      direction="row"
      item={true}
      spacing={0}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <Component
          className="user-mock"
          src="test-file-stub"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(Typography))
            className="username-mock"
            variant="h5"
          >
            123
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(Typography))
            className="address-mock"
            variant="caption"
          >
            123
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      className="actions-mock"
      container={true}
      direction="column"
      item={true}
      justify="space-between"
    >
      <WithStyles(ForwardRef(Grid))
        className="pointer-mock"
        container={true}
        item={true}
        onClick={[Function]}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <Component
            className="removeIcon-mock"
            src="test-file-stub"
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={true}
        >
          <WithStyles(ForwardRef(Typography))
            className="action-mock"
            variant="body1"
          >
            Hide message
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="banDiv-mock"
        container={true}
        item={true}
        onClick={[Function]}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <Component
            className="banIcon-mock"
            src="test-file-stub"
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={true}
        >
          <WithStyles(ForwardRef(Typography))
            className="action-mock"
            variant="body1"
          >
            Silence user
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</ForwardRef(Popper)>
`;
